docs/docs/contribution/i18n.md

```diff
@@ -0,0 +1,209 @@
+  isHTML: true,
+  interpol: "abc"
+})
+
+window.deps.translateString(
+  'core.login.successWithName',
+  'Logged in as: {{name}}',
+  {
+      name: displayName,
+  },
+)
+```
+
+The function is completely type-safe:
+
+- The first argument is `key`, which is validated with valid keys in `i18n/locales/en/`.
+- The second argument is `value`. With the `key` provided, the type of `value` will be narrowed down to the value in the translation file to ensure all occurrences of the key have the same value. If this is an HTML string (with the `isHTML` attribute set to `true`), the value will automatically be abstracted when exporting the translation.
+- The third argument is `interpolation`, which could be omitted when the value doesn't contain any interpolation. Otherwise, it should be an object whose attributes are inferred from the `value`. Moreover, an additional `isHTML` boolean attribute is added to indicate whether the `value` contains HTML tags and should be abstracted when exporting the translation.
+
+To provide better readability and maintainability, only string-level translation will be used in .ts files when it comes to dynamic content. For static content, element-level translation is still the best choice.
+
+### Scripts
+Several npm scripts are available to facilitate the i18n workflow:
+
+- [`vscode-intellisense`](../../../scripts/vscode-intellisense.js): Generates the `html.html-data.json` file in the `.vscode` folder to enhance intellisense for the `data-i18n` attribute in HTML files in VSCode.
+- [`i18n-export`](../../../scripts/i18n-export.js): Extracts source texts from the codebase and generates `.ts` and `.lokalise.json` files under `src/livecodes/i18n/locales/en`.
+  - The `--save-tmp` flag can be used to save the extracted source texts to `i18n/locales/tmp` instead of `en` for debugging purposes.
+- [`i18n-update-push`](../../../scripts/i18n-upload.mjs): **(Only used in CI and should not be run locally)** Pushes the source texts in `.lokasile.json` files to Lokalise.
+  - The `<branch>` argument is required to specify the branch to push to.
+- [`i18n-update-pull`](../../../scripts/i18n-import.mjs): **(Only used in CI and should not be run locally)** Pulls the translated texts from Lokalise and updates the `.ts` files under `src/livecodes/i18n/locales`. *Outdated translation will be deprecated during import.*
+  - The `<branch>` argument is required to specify the branch to pull from.
+- [`i18n-exclude`](../../../scripts/i18n-exclude.js): **(Only used in other scripts and should not be run locally)** Excludes all other i18n locales except for English from type checking, as they might stay outdated and cause errors.
+  - The `<phase>` argument is required to specify whether the script is to exclude files or revert the exclusion. Valid values are `pre` and `post`.
+
+Please run `i18n-export` before pushing changes to the codebase to ensure that the source texts are up-to-date.
+
+## For Translators
```

Comment:
Fixed in https://github.com/zyf722/livecodes/commit/a2b69dc.

> We can also put a link in the repo README to the project on Lokalise

Are you interested in the solution I propose [here](https://github.com/orgs/live-codes/discussions/547#discussioncomment-9973868)?

Reply:
> Are you interested in the solution I proposed [here](https://github.com/orgs/live-codes/discussions/547#discussioncomment-9973868)?

Yes, very much 👍 
Please advise how we can take this forward.

Reply:
Just fork [zyf722/lokalise-badge](https://github.com/zyf722/lokalise-badge), add `LOKALISE_API_TOKEN` and `LOKALISE_PROJECT_ID` to repository secrets, and then we can reference it using `https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/live-codes/lokalise-badge/main/badge.json`.

It might be better to have a special bot account on Lokalise using a livecodes (dot) io email address, and apply the necessary permissions to it, so that we can easily manage the API token (for this badge workflow and other workflows in the main repo).

Reply:
Done
https://github.com/live-codes/lokalise-badge

[![Badge](https://img.shields.io/endpoint?url=https%3A%2F%2Fraw.githubusercontent.com%2Flive-codes%2Flokalise-badge%2Fmain%2Fbadge.json)](https://app.lokalise.com/public/34958094667a72e9454592.95108106/)

Reply:
Fixed in https://github.com/zyf722/livecodes/commit/.f62e41b.

Reply:
> Done https://github.com/live-codes/lokalise-badge

It seems that the workflow is not running as expected, due to the fact that scheduled workflows are disabled by default when a public repository is forked. You might need to manually enable it to activate the workflow and cron trigger.

Related docs here:
https://docs.github.com/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/disabling-and-enabling-a-workflow